packages/discovery-react-components/src/components/DocumentPreview/DocumentPreview.tsx

```diff
@@ -54,6 +54,10 @@ interface Props extends WithErrorBoundaryProps {
    * React component rendered as a fallback when no preview is available
    */
   fallbackComponent?: ComponentProps<typeof SimpleDocument>['fallbackComponent'];
+  /**
+   * Callback to receive changes in document preview state
+   */
+  onPreviewStateChange?: (state: { currentPage?: number }) => void;
```

Comment:
I choose generic name rather than `onCurrentPageChange` so that we can easily add events to be notified.

Reply:
In that case, should we just call this `onChange`? Using "state" is a bit redundant, and we already know it's the "preview" component.